src/JNI.jl

```diff
@@ -0,0 +1,767 @@
+module JNI
+
+# jnienv.jl exports
+export JNINativeInterface, JNIEnv, JNIInvokeInterface, JavaVM
+# jni_md.h exports
+export jint, jlong, jbyte
+# jni.h exports
+export jboolean, jchar, jshort, jfloat, jdouble, jsize, jprimitive
+# constant export
+export JNI_TRUE, JNI_FALSE
+# Legacy exports
+export jnifunc
+
+include("jnienv.jl")
+
+global jnifunc
```

Comment:
maybe we should make these globals `Ref`s now? JavaCall started when that feature did not exist in Julia, but maybe its worth doing now?

Reply:
Done with b50fc15